pkg/anypoint/authentication.go

```diff
@@ -6,37 +6,40 @@ import (
 	"github.com/Axway/agent-sdk/pkg/util/log"
 )
 
-// Auth represents the authentication information.
+// Auth gets a token and starts a refresh interval.
 type Auth interface {
+	Start() error
 	Stop()
 	GetToken() string
 	GetOrgID() string
 }
 
-// auth represents the authentication information.
 type auth struct {
 	token    string
 	user     *User
-	client   AuthClient
+	client   TokenClient
 	stopChan chan struct{}
+	done     bool
 }
 
-// NewAuth creates a new authentication token
-func NewAuth(client AuthClient) (Auth, error) {
-	a := &auth{
+// NewAuth gets an access token
+func NewAuth(client TokenClient) *auth {
+	return &auth{
 		stopChan: make(chan struct{}),
+		client:   client,
 	}
-	token, user, lifetime, err := client.GetAccessToken()
```

Comment:
Having this in the constructor made testing difficult so I moved it to a new method.